samples/system-test/detect.v2beta1.test.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2018, Google, LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const test = require(`ava`);
+const {runAsync} = require('@google-cloud/nodejs-repo-tools');
+
+const cmd = 'node detect.v2beta1.js';
+//const testQuery = [`How do I sign up?`];
+const testQuery = `Where is my data stored?`;
+
+const testKnowledgeBaseName = 'TestKnowBase';
+
+let knowbaseDisplayName;
+let knowbaseFullName;
+
+let documentFullPath;
+const testDocName = `TestDoc`;
+//const testDocumentPath = `https://dialogflow.com/docs/concepts/google-projects-faq`;
+const testDocumentPath = `https://cloud.google.com/storage/docs/faq`;
+
+test.before(`It should create a knowledge base`, async t => {
+  const output = await runAsync(
+    `${cmd} createKnowledgeBase -k ${testKnowledgeBaseName}`
+  );
+  const parsedOut = output.split(`\n`);
```

Comment:
❗️ **#BLOCKER**

Reply:
This is to grab document id for deletion in next test step- please advice, cannonicals use this same method, see ln 72 here: https://github.com/dialogflow/dialogflow-python-client-v2/blob/master/samples/tests/knowledge_base_test.py

Reply:
```js
test.before(`It should create a knowledge base`, async t => {
  let output = await runAsync(
    `${cmd} listKnowledgeBases`
  );
  t.false(output.includes(testKnowledgeBaseName));

  output = await runAsync(
    `${cmd} createKnowledgeBase -k ${testKnowledgeBaseName}`
  );
  t.true(output.includes(`Name: ${testKnowledgeBaseName}`));
  t.true(output.includes(`Display Name: ${testKnowledgeBaseDisplayName}`));

  output = await runAsync(
    `${cmd} listKnowledgeBases`
  );
  t.true(output.includes(testKnowledgeBaseName));
});
```

----

Unabridged with descriptions

```js
test.before(`It should create a knowledge base`, async t => {
  // Negative Assertion (confirm the 'before state' – make sure this KB doesn't already exists)
  let output = await runAsync(
    `${cmd} listKnowledgeBases`
  );
  t.false(output.includes(testKnowledgeBaseName));

  // Run the code! Do the thing!
  output = await runAsync(
    `${cmd} createKnowledgeBase -k ${testKnowledgeBaseName}`
  );
  t.true(output.includes(`Name: ${testKnowledgeBaseName}`)); // Now we verify the exact output we care about
  t.true(output.includes(`Display Name: ${testKnowledgeBaseDisplayName}`));

  // Positive Assertion (confirm the 'after state' – make sure this KB properly exists now)
  //
  // To be as flexible as possible (read: not brittle) ALL we test for is the KB name
  //    there are other tests to make sure that listKnowledgeBases prints out in a certain format/etc
  //    we don't want this test to break if that changes
  output = await runAsync(
    `${cmd} listKnowledgeBases`
  );
  t.true(output.includes(testKnowledgeBaseName));
});
```

Reply:
WHOA. I mean `test.serial`. Whoa. Somehow this has `test.before`. Suuuuper wrong. Lol, please update.

\**edits the code samples above to be `test.serial` instead of `test.before` which is used for setup, not actual tests*\*